Add of Migration Trials and initial refactoring of Ecommerce Trial components

zaerl opened the pull request:
This initial commit will cover the addition of Migration Trial and refactoring of WooExpress Trials to become generic "trial" components.

The PR does not cover 100% of components to keep the PR size down. The next changes are on https://github.com/Automattic/wp-calypso/pull/80141 (draft).

This is the list of the components that are used:

1. `CurrentPlan`
    - `ECommerceTrialCurrentPlan`
       - `ECommerceTrialBanner`
       - List of features included
       - Ready to start selling?
       - List of features not included
2. `ECommerceTrialExpired` (not changed right now)
       - `TrialUpgradeConfirmation` (not changed right now)
3. `Plans`
     - `ECommerceTrialPlansPage` (not changed right now)

Additional: p1690534487000689-slack-C01A60HCGUA

Fixes https://github.com/Automattic/wp-calypso/issues/80076
Fixes https://github.com/Automattic/wp-calypso/issues/79955

## Proposed Changes

* Move the trial checkout URL functions to a generic `client/lib/trials/get-*-trial-checkout-url.ts`
* Add support in the `CurrentPlan` component for `PLAN_MIGRATION_TRIAL_MONTHLY` plans

New components and refactoring:
1. `CurrentPlan`
    - `TrialCurrentPlan` <- moved to `client/my-sites/plans/current-plan/trials/trial-current-plan.tsx` and renamed
       - `TrialBanner`
       - Replace the list with new components `EcommerceTrialIncluded`, `MigrationTrialIncluded`
       - Ready to start selling?
       - Replace the list with new components `EcommerceTrialNotIncluded`, `MigrationTrialNotIncluded`
4. `ECommerceTrialExpired` (not changed right now)
      - `TrialUpgradeConfirmation` (not changed right now)
5. `Plans`
     - `ECommerceTrialPlansPage` (not changed right now)

## Notes
1. The migration current plan page show buttons in the boxes. In the ECommerce one, they are hidden by default.
2. We have no additional section similar to the "Ready to start selling?" on the ECommerce page. So the Migration page right now shows nothing.
3. "Jetpack backups and restores" CTA is TBD now. I've used "View your backup activity" for now (not localized).

These are the URLs that are right now associated with the CTAs:

1. "Beautiful Themes: **Browse themes**":  `/themes/__URL__`
2. "Advanced Design Tools: **Design your blog**": `wp-admin/site-editor.php`
5. "Newsletter: **Setup a newsletter**":  `/setup/newsletter/intro?siteSlug=__SLUG__`
6. "Jetpack backups and restores: **View your backup activity**": `/backup/__SLUG__`
7. "Spam Protection: **Keep your site safe**": `https://jetpack.com/blog/what-is-spam/`
8. "SEO tools: **Increase visibility**": `https://wordpress.com/support/seo-tools/`
9. "Google Analytics: **Connect Google Analytics**": `https://wordpress.com/support/google-analytics/`

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Create a JN blog, install "Move to WP" and from `localhost` and start the free plan
* Create a blog from WooCommerce site and choose a free plan
* Visit `http://calypso.localhost:3000/plans/my-plan/__JN_URL__`
* Visit `http://calypso.localhost:3000/plans/my-plan/__WOOEXPRESS_URL__`
* Visit `https://wordpress.com/plans/my-plan/__WOOEXPRESS_URL__`

* Ensure that the Migration Trial page reflects the [Figma file](https://href.li/?https://www.figma.com/file/nRzixgCz1TsnOLPiKr4awj/Free-trials-in-the-Migrate-to-WP.com-flow?type=design&node-id=1-64&mode=design&t=SHxI569VmRWJ1Aaf-0)
* Ensure that the WooCommerce page is the same found in production

![screenshot](https://github.com/Automattic/wp-calypso/assets/167611/8254a31c-df61-4b1d-87ef-96277472ebb3)

## ECommerce final part
<img width="1077" alt="image" src="https://github.com/Automattic/wp-calypso/assets/167611/b41c839c-6f49-4665-a900-99a0857cb14e">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

zaerl committed:
Move trial checkout to separate directory

zaerl committed:
Fix: wrong imports

zaerl committed:
Move feature list to separate components

zaerl committed:
Rename ECommerceTrialCurrentPlan to TrialCurrentPlan

zaerl committed:
Rename component CSS classes

zaerl committed:
Add migration what's included component

zaerl committed:
Fix: add newsletter SVG

zaerl committed:
Add CTA links

zaerl committed:
Update newsletters icon

zaerl committed:
Fix: change Akismet URL

zaerl committed:
Fix: change newsletter URL

ouikhuan commented on client/lib/trials/get-migration-trial-checkout-url.ts:
```diff
@@ -0,0 +1,16 @@
+	siteSlug: string;
+}
+
+export function getMigrationTrialCheckoutUrl( {
+	productSlug,
+	siteSlug,
+}: MigrationTrialCheckoutUrlArguments ): string {
+	return addQueryArgs(
+		{ redirect_to: `/plans/my-plan/trial-upgraded/${ siteSlug }` },
+		`/checkout/${ siteSlug }/${ productSlug }`
+	);
+}
```
Since this function is nearly identical to the `get-ecommerce-trial-checkout-url.ts`, should we make this a more general trial function for trial upgrades? Maybe something like `getTrialCheckoutUrl`? Or do you see this will be different compare to the eCommerceTrial one in the near future?

zaerl replied:
I created a new one because there will probably (?) be something different. But we can use one for now.

ouikhuan commented on client/my-sites/plans/current-plan/trials/trial-current-plan.tsx:
```diff
@@ -27,18 +31,28 @@ const TrialCurrentPlan = () => {
 	const displayAllIncluded = ! isMobile || showAllTrialFeaturesInMobileView;
+	const bodyClass = isEcommerceTrial
+		? [ 'is-ecommerce-trial-plan' ]
+		: [ 'is-migration-trial-plan' ];
```
Consider of future hosting trial, should we make this as a `is-bussiness-trial-plan` instead?

zaerl replied:
Sure, good idea.

ouikhuan commented on client/my-sites/plans/current-plan/trials/migration-trial-included.tsx:
```diff
@@ -0,0 +1,108 @@
+	translate: typeof translate;
+	displayAll: boolean;
+}
+const MigrationTrialIncluded: FunctionComponent< Props > = ( props ) => {
```
Maybe for this section it's better to make it as a `BusinessTrialIncluded` as well, what do you think?

ouikhuan commented on client/my-sites/plans/current-plan/trials/migration-trial-included.tsx:
```diff
@@ -0,0 +1,108 @@
+const MigrationTrialIncluded: FunctionComponent< Props > = ( props ) => {
+	const { translate, displayAll = true } = props;
+
+	// TODO: translate when final copy is available
```
I think we can get rid of this line

ouikhuan commented on client/my-sites/plans/current-plan/trials/trial-current-plan.tsx:
```diff
@@ -0,0 +1,139 @@
+	 *
+	 * @param ctaPosition - The position of the CTA that triggered the redirect.
+	 */
+	const goToCheckoutWithPlan = ( ctaPosition: string ) => {
```
Currently if I click the upgrade plan button at the bottom of the page in a migration trial plan. It gives me an error of `Sorry, that plan is not a valid upgrade from your current plan.` on the checkout page.

zaerl committed:
Minor refactoring of checkout URL

zaerl committed:
Fix: rename CSS class

zaerl committed:
Rename migration trial folder

zaerl committed:
Minor components name refactoring

zaerl commented:
@ouikhuan 

Apply: D118096-code

1. Renamed `getECommerceTrialCheckoutUrl` to `getTrialCheckoutUrl` and removed `getMigrationTrialCheckoutUrl`
2. Renamed `.is-migration-trial-plan` CSS class to `.is-business-trial-plan`
3. Renamed `calypso/assets/images/plans/wpcom/migration-trial/` folder to `calypso/assets/images/plans/wpcom/business-trial/`
4. Fix: "Sorry, that plan is not a valid upgrade from your current plan."

zaerl committed:
Fix: changed default plan

ouikhuan reviewed (approved):
LGTM! 🚢

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8682888

Thank you @zaerl for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8682888).
